.github/workflows/build-and-test.yml

```diff
@@ -20,7 +20,7 @@ jobs:
         include:
           - os: ubuntu-22.04
             target: x86_64-unknown-linux-gnu
-            channel: 1.63
+            channel: 1.85
```

Comment:
This is the minimum supported Rust version. I'd be inclined to keep this as low as possible unless using features that are only supported in new Rust versions.

It should also be kept in sync with the `rust-version` field in the two `Cargo.toml` files.

Reply:
I agree that this should be kept at the lowest possible version that works.   My preference would be :

1) Keep the Cargo.toml in sync with the lowest version we can.

2) Configure the github workflows to test against each of : MSRV, default and nightly.

Reply:
Thanks @ebarnard , @goulart-paul - I've reverted the changes to `rust-version` to be the same as before (1.63). Turns out `bindgen` was spitting out bindings that wasn't compatible with 1.63, so I've added `--rust-target 1.63` to the `generate_bindings.sh` as well.

I've added some missing targets in the CI so its checking on 1.63/stable/beta/nightly channels on all targets now.